packages/test/src/workflows/smorgasbord.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * This workflow does a little bit of everything
+ */
+import {
+  Trigger,
+  sleep,
+  createChildWorkflowHandle,
+  createActivityHandle,
+  ActivityCancellationType,
+  CancellationScope,
+  isCancellation,
+} from '@temporalio/workflow';
+import { Smorgasbord } from '../interfaces';
+import * as activities from '../activities/';
+import { signalTarget } from './signal-target';
+
+const { fakeProgress } = createActivityHandle<typeof activities>({
+  startToCloseTimeout: '5s',
+  scheduleToCloseTimeout: '10s',
+  heartbeatTimeout: '3s',
+  cancellationType: ActivityCancellationType.WAIT_CANCELLATION_COMPLETED,
+});
+
+export const smorgasbord: Smorgasbord = () => {
+  let step = 0;
+  const unblocked = new Trigger<void>();
+
+  return {
+    queries: {
+      step(): number {
+        return step;
+      },
+    },
+    signals: {
+      activityStarted(): void {
+        unblocked.resolve();
+      },
+    },
+
+    async execute(): Promise<void> {
+      while (step < 2) {
```

Comment:
```suggestion
      for (; step < 2; ++step) {
```
And check if step `=== 1` below